apps/api/src/app/inbox/usecases/get-preferences/get-preferences.usecase.ts

```diff
@@ -0,0 +1,69 @@
+import { Injectable, NotFoundException } from '@nestjs/common';
+import {
+  AnalyticsService,
+  GetSubscriberGlobalPreference,
+  GetSubscriberGlobalPreferenceCommand,
+  GetSubscriberWorkflowPreference,
+  GetSubscriberWorkflowPreferenceCommand,
+} from '@novu/application-generic';
+import { NotificationTemplateRepository, SubscriberRepository } from '@novu/dal';
+import { ISubscriberPreferences, PreferenceLevelEnum } from '@novu/shared';
+import { GetPreferencesCommand } from './get-preferences.command';
+
+@Injectable()
+export class GetPreferences {
+  constructor(
+    private subscriberRepository: SubscriberRepository,
+    private notificationTemplateRepository: NotificationTemplateRepository,
+    private getSubscriberWorkflowPreferenceUsecase: GetSubscriberWorkflowPreference,
+    private getSubscriberGlobalPreference: GetSubscriberGlobalPreference,
+    private analyticsService: AnalyticsService
+  ) {}
+
+  async execute(command: GetPreferencesCommand): Promise<ISubscriberPreferences[]> {
+    const subscriber = await this.subscriberRepository.findBySubscriberId(command.environmentId, command.subscriberId);
+
+    if (!subscriber) {
+      throw new NotFoundException(`Subscriber: ${command.subscriberId} not found`);
+    }
+
+    const globalPreference = await this.getSubscriberGlobalPreference.execute(
+      GetSubscriberGlobalPreferenceCommand.create({
+        organizationId: command.organizationId,
+        environmentId: command.environmentId,
+        subscriberId: command.subscriberId,
+      })
+    );
+
+    const updatedGlobalPreference = {
+      level: PreferenceLevelEnum.GLOBAL,
+      preferences: globalPreference.preference,
```

Comment:
<!-- Notes are always non-blocking and simply highlight something the reader should take note of.  -->
🗒 **note (non-blocking):** In the DX guide we've missed the `enabled` flag, can you please update it?
```
{
      level: PreferenceLevelEnum.GLOBAL,
      preferences: { // <--- the correct shape of preferences
           enabled,
           channels,
      }
}
```

Reply:
![image](https://github.com/user-attachments/assets/2c3f23db-0596-4cb0-9476-e6477b2387ac)
It contains the `enabled` flag

Reply:
![image](https://github.com/user-attachments/assets/2c3f23db-0596-4cb0-9476-e6477b2387ac)
It contains the `enabled` flag

Reply:
I mean it's missing in the guide ;) https://www.notion.so/novuhq/The-new-Inbox-REST-API-efa2c9b4592a4f5e9b20011f9202a7c3?pvs=4#bf5c6344cc724132a04cf881e03215b1